mrp_production_split/models/mrp_production.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2023 Camptocamp SA (https://www.camptocamp.com).
+# @author Iván Todorovich <ivan.todorovich@camptocamp.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    def action_split(self):
+        self.ensure_one()
+        self._check_company()
+        if self.state in ("draft", "done", "to_close", "cancel"):
+            raise UserError(
+                _("Cannot split a manufacturing order that is done or cancelled.")
+            )
```

Comment:
if can't split in done or canceled, why draft is in states?

Reply:
Oops, I forgot to update the message string.
It's changed now ;) thanks!